web/client/components/Search/SearchBar.jsx

```diff
@@ -73,11 +76,20 @@ let SearchBar = React.createClass({
         //  const innerGlyphicon = <Button onClick={this.search}></Button>;
         const remove = <Glyphicon className="searchclear" glyph="remove" onClick={this.clearSearch}/>;
         var showRemove = this.state.searchText !== "";
+        let placeholder = this.props.placeholder;
+        let placeholderLocMessage = LocaleUtils.getMessageFromMessageComponent(placeholder);
```

Comment:
Please, add contextTypes to the component, so that you can access localized messages directly from here.